src/azure-cli/azure/cli/command_modules/acs/_validators.py

```diff
@@ -392,3 +392,12 @@ def validate_label(label):
                        "characters, '-', '_' or '.', and must start and end with an alphanumeric character" % label)
 
     return {kv[0]: kv[1]}
+
+
+def validate_assign_identity(namespace):
+    if namespace.assign_identity is not None:
+        if namespace.assign_identity == '':
+            return
+        from msrestazure.tools import is_valid_resource_id
+        if not is_valid_resource_id(namespace.assign_identity):
+            raise CLIError("--assign-identity is not a valid Azure resource ID.")
```

Comment:
Can you use [`InvalidArgumentValueError`](https://github.com/Azure/azure-cli/blob/7b927fe3a01e25377571532ec22c48425bdc8a62/src/azure-cli-core/azure/cli/core/azclierror.py#L123) here?